dev_guide/builds.adoc

```diff
@@ -1577,3 +1577,69 @@ Docker and Source builds set the following labels on output images:
 |*io.openshift.build.source-location*
 |Source URL for the build
 |===
+
+
+[[using-external-artifacts]]
+= Using External Artifacts During a Build
+
+It is not recommended to store binary files in a source repository. Therefore
+you may find it necessary to define a build which pulls additional files (such
+as Java jar dependencies) during the build process. How this is done depends on
+the build strategy you are using:
+
+For a `*Source*` build strategy you need to put appropriate shell commands into
+the `assemble` script.
+
+.`.sti/bin/assemble` file
+====
+
+[source,bash]
+----
+#!/bin/sh
+APP_VERSION=1.0
+wget http://repository.example.com/app/app-$APP_VERSION.jar -O app.jar
+----
+====
+
+.`.sti/bin/run` file
+====
+
+[source,bash]
+----
+#!/bin/sh
+exec java -jar app.jar
+----
+====
+
+For more information on how to control which assemble and run script is used by a source bulid,
+see: link:#override-builder-image-scripts[Override Builder Image Scripts]
+
+For a `*Docker*` build strategy you need to modify the `Dockerfile` and invoke
+shell commands with the https://docs.docker.com/engine/reference/builder/#run[`RUN` instruction].
+
+.Excerpt of `Dockerfile`
+====
+
+[source]
+----
+FROM jboss/base-jdk:8
+
+ENV APP_VERSION 1.0
+RUN wget http://repository.example.com/app/app-$APP_VERSION.jar -O app.jar
+
+EXPOSE 8080
+CMD [ "java", "-jar", "app.jar" ]
+----
+====
+
+In practice you may want to use an environment variable for the file location
+so that the specific file to be downloaded can be customized via an environment
+variable defined on the `BuildConfig` rather than updating the assemble script or
+`Dockerfile`.
+
+You could choose between a different ways of defining environment variables:
+
+- link:#environment-files[put into the `.sti/environment` file] (only for a `*Source*` build strategy)
+- link:#buildconfig-environment[set in `*BuildConfig*`]
+- link:../cli_reference/basic_cli_operations.html#build-and-deployment-cli-operations[provide explicitly with `*oc start-build --env*`]
+  (only for builds that are triggered manually)
```

Comment:
I would suggest reordering this list so that the first item is the one that applies in all cases:

* set in BuildConfig
* set in .sti/environment
* set in start-build

Obviously the env can also be set and updated in the Dockerfile itself, but I imagine that would be a dumb idea.